Source/ExtendedStorage/WorkGiver_ExtendedStorage.cs

```diff
@@ -0,0 +1,70 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using RimWorld;
+using Verse;
+using Verse.AI;
+
+namespace ExtendedStorage
+{
+    public class WorkGiver_Merge: WorkGiver_Scanner
+    {
+        public override IEnumerable<Thing> PotentialWorkThingsGlobal( Pawn pawn )
+        {
+            return pawn.Map.listerBuildings.AllBuildingsColonistOfClass<Building_ExtendedStorage>()
+                .Where( b => !b.AtCapacity 
+                        && !b.IsForbidden( pawn )
+                        && b.StoredThingDef != null 
```

Comment:
can be simplified to `StoredThingTotal>0` since `StoredThingTotal` implicitly checks if `StoredThingDef`is null and just returns `0`